Hrychyna.University.InteractivePlantCareAssistant.Web/script.js

```diff
@@ -0,0 +1,133 @@
+let currentThread = null;
+
+function addThread() {
+    var input = document.getElementById('newThreadTitle');
+    var threadTitle = input.value.trim();
+    if (threadTitle === "") {
+        alert("Please enter a topic name.");
+        return;
+    }
+
+    var threadsList = document.getElementById('threadsList');
+    var li = document.createElement('li');
+
+    var userName = "Dariia";
+    var currentTime = new Date().toLocaleString("uk-UA");
+
+    li.innerHTML = `<span class="thread-title">${threadTitle}</span> <span class="thread-info">(Created by ${userName} at ${currentTime})</span>
+        <span class="likes" onclick="toggleLike(this)">
+            <svg class="heart" viewBox="0 0 32 29.6">
+                <path d="M23.6,0A8.6,8.6,0,0,1,32,8.6c0,9.1-14.4,21-16,21S0,17.7,0,8.6A8.6,8.6,0,0,1,8.6,0a9.3,9.3,0,0,1,7.4,3.6A9.3,9.3,0,0,1,23.6,0Z"/>
+            </svg>
+            <span class="like-count">0</span>
+        </span>`;
+
+    li.onclick = function(event) {
+        if (event.target.closest('.likes')) return;
```

Comment:
Check if event.target.closest('.likes') is truthy in an if statement directly.

Reply:
Thanks for the heads up, I'll look into that suggestion.